src/main/java/com/internet/shop/dao/jdbc/UserDaoJdbcImpl.java

```diff
@@ -0,0 +1,200 @@
+            if (resultSet.next()) {
+                user = getUserFromResultSet(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not get user from DB with login = " + login, e);
+        }
+        user.setRoles(getRolesOfUser(user.getId()));
+        return Optional.ofNullable(user);
+    }
+
+    @Override
+    public User create(User item) {
+        String query = "INSERT INTO users(name, login, password) VALUES(?, ?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, item.getName());
+            statement.setString(2, item.getLogin());
+            statement.setString(3, item.getPassword());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                item.setId(resultSet.getLong(1));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to create user " + item, e);
+        }
+        return defineRoles(item);
+    }
+
+    @Override
+    public Optional<User> getById(Long item) {
+        User user = new User();
+        String query = "SELECT * FROM users "
+                + "INNER JOIN users_roles ON users.user_id = users_roles.user_id "
+                + "WHERE users.deleted = false AND users.user_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, item);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                user = getUserFromResultSet(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to create user with ID " + item, e);
+        }
+        user.setRoles(getRolesOfUser(user.getId()));
+        return Optional.ofNullable(user);
+    }
+
+    @Override
+    public User update(User item) {
+        String query = "UPDATE users SET name = ?, login = ?, password = ? "
+                + "WHERE user_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(1, item.getName());
+            statement.setString(2, item.getLogin());
+            statement.setString(3, item.getPassword());
+            statement.setLong(4, item.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to create user " + item, e);
+        }
+        deleteRoles(item.getId());
+        defineRoles(item);
+        item.setRoles(getRolesOfUser(item.getId()));
```

Comment:
You've already set roles in the defineRoles method

Reply:
The same.